data/tutorials/language/3ds_02_maps.md

```diff
@@ -0,0 +1,307 @@
+---
+id: maps
+title: Maps
+description: >
+  Create a mapping using the standard library's Map module
+category: "Data Structures"
+---
+
+## Introduction
+
+The [`Map`](/api/Map.html) module lets you create _immutable_ key-value [association
+tables](https://en.wikipedia.org/wiki/Associative_array) for your types. Such
+maps are never modified, and every operation returns a new map instead.
+
+**Note**: Maps described in this tutorial should not be confused with map
+functions such as `List.map`, `Array.map`, `Option.map` and others. The maps
+described in this tutorial are also called dictionaries or associative tables.
+
+To use `Map`, we first have to use the [`Map.Make`](/api/Map.Make.html) functor to create our custom
+map module. Refer to the [Functors](/docs/functors) for more information on
+functors. This functor has a module parameter that defines the keys' type to
+be used in the maps, and a function for comparing them.
+```ocaml
+# module StringMap = Map.Make(String);;
+
+module StringMap :
+  sig
+    type key = string
+    type 'a t = 'a Map.Make(String).t
+    val empty : 'a t
+    val add : key -> 'a -> 'a t -> 'a t
+    val add_to_list : key -> 'a -> 'a list t -> 'a list t
+    val update : key -> ('a option -> 'a option) -> 'a t -> 'a t
+    val singleton : key -> 'a -> 'a t
+    val remove : key -> 'a t -> 'a t
+    (* ... *)
+  end
+```
+
+After naming the newly-created module `StringMap`, OCaml's toplevel displays the
+module's signature. Since it contains a large number of functions, the output
+copied here is shortened for brevity `(...)`.
+
+This module doesn't define the values' type. It will be defined when we create
+our first map.
+
+## Creating a Map
+
+The `StringMap` module has an `empty` value that has a type parameter `'a` in
+its type: `empty : 'a t`.
+
+This means that we can use `empty` to create new empty maps where the value is of any type.
+```ocaml
+# let int_map : int StringMap.t = StringMap.empty;;
+val int_map : int StringMap.t = <abstr>
+
+# let float_map : float StringMap.t = StringMap.empty;;
+val float_map : float StringMap.t = <abstr>
+```
+
+The type of the values can be specified in two ways:
+* When you create your new map with an annotation
+* By adding an element to the map:
+```ocaml
+# let int_map = StringMap.(empty |> add "one" 1);;
+val int_map : int StringMap.t = <abstr>
+```
+
+## Working With Maps
+
+Throughout the rest of this tutorial, we use the following map:
+```ocaml
+# let lucky_numbers = StringMap.of_list [
+    ("leostera", 2112);
+    ("charstring88", 88);
+    ("divagnz", 13);
+  ];;
+val lucky_numbers : int StringMap.t = <abstr>
```

Comment:
That's an OCaml version issue. `of_list` was added in OCaml 5.1. We can have this to make compatible with 4.14
```suggestion
let lucky_numbers = StringMap.of_seq @@ List.to_seq [
```

BTW. Should it be a requirement for all tutorials? Be compatible with 4.14 ? Since it is LTS it probably make sense. 

We can have 5.x related stuff as remarks/notes